src/brief.jl

```diff
@@ -148,7 +148,11 @@ function centered(size::Int, window::Int, seed::Int)
 end
 
 function create_descriptor{T<:Gray}(img::AbstractArray{T, 2}, keypoints::Keypoints, params::BRIEF)
-    img_smoothed = imfilter_gaussian(img, [params.sigma, params.sigma])
+    
+    _eltype{T,C<:Colorant}(::Type{T}, ::Type{C}) = base_colorant_type(C){T}
+    _eltype{T,R<:Real}(::Type{T}, ::Type{R}) = T
+    factkernel = KernelFactors.IIRGaussian([params.sigma, params.sigma])
+    img_smoothed = imfilter(_eltype(Float64, eltype(img)), img, factkernel, NA())
```

Comment:
Yes, the eltype is correctly chosen so we don't need the `_eltype` function to do that for us. I will modify the code as required.